runtime/integration-tests/src/generic/cases/liquidity_pools.rs

```diff
@@ -0,0 +1,2505 @@
+				additional: CustomMetadata {
+					transferability: CrossChainTransferability::Xcm(Default::default()),
+					..CustomMetadata::default()
+				},
+			};
+
+			assert_ok!(orml_asset_registry::Pallet::<T>::register_asset(
+				<T as frame_system::Config>::RuntimeOrigin::root(),
+				meta,
+				Some(CurrencyId::Native)
+			));
+		}
+
+		pub fn register_ksm<T: Runtime + FudgeSupport>() {
+			let meta: AssetMetadata<Balance, CustomMetadata> = AssetMetadata {
+				decimals: 12,
+				name: "Kusama".into(),
+				symbol: "KSM".into(),
+				existential_deposit: 1_000_000_000,
+				location: Some(VersionedMultiLocation::V3(MultiLocation::new(1, Here))),
+				additional: CustomMetadata {
+					transferability: CrossChainTransferability::Xcm(Default::default()),
+					..CustomMetadata::default()
+				},
+			};
+
+			assert_ok!(orml_asset_registry::Pallet::<T>::register_asset(
+				<T as frame_system::Config>::RuntimeOrigin::root(),
+				meta,
+				Some(KSM_ASSET_ID)
+			));
+		}
+
+		pub fn air(amount: Balance) -> Balance {
+			amount * dollar(currency_decimals::NATIVE)
+		}
+
+		pub fn ausd(amount: Balance) -> Balance {
+			amount * dollar(currency_decimals::AUSD)
+		}
+
+		pub fn ksm(amount: Balance) -> Balance {
+			amount * dollar(currency_decimals::KSM)
+		}
+
+		pub fn foreign(amount: Balance, decimals: u32) -> Balance {
+			amount * dollar(decimals)
+		}
+
+		pub fn dollar(decimals: u32) -> Balance {
+			10u128.saturating_pow(decimals)
+		}
+
+		pub fn air_fee() -> Balance {
+			fee(currency_decimals::NATIVE)
+		}
+
+		pub fn ausd_fee() -> Balance {
+			fee(currency_decimals::AUSD)
+		}
+
+		pub fn fee(decimals: u32) -> Balance {
+			calc_fee(default_per_second(decimals))
+		}
+
+		// The fee associated with transferring KSM tokens
+		pub fn ksm_fee() -> Balance {
+			calc_fee(ksm_per_second())
+		}
+
+		pub fn calc_fee(fee_per_second: Balance) -> Balance {
+			// We divide the fee to align its unit and multiply by 4 as that seems to be the
+			// unit of time the tests take.
+			// NOTE: it is possible that in different machines this value may differ. We
+			// shall see.
+			fee_per_second.div_euclid(10_000) * 8
+		}
+	}
+
+	use utils::*;
+
+	mod restricted_calls {
+		use super::*;
+
+		fn xtokens_transfer<T: Runtime + FudgeSupport>() {
+			let mut env = FudgeEnv::<T>::from_storage(Default::default(), Default::default());
+
+			env.parachain_state_mut(|| {
+				assert_noop!(
+					orml_xtokens::Pallet::<T>::transfer(
+						RawOrigin::Signed(Keyring::Alice.into()).into(),
+						CurrencyId::Tranche(401, [0; 16]),
+						42,
+						Box::new(
+							MultiLocation::new(
+								1,
+								X2(
+									Parachain(T::FudgeHandle::SIBLING_ID),
+									Junction::AccountId32 {
+										network: None,
+										id: Keyring::Bob.into(),
+									}
+								)
+							)
+							.into()
+						),
+						WeightLimit::Limited(8_000_000_000_000.into()),
+					),
+					orml_xtokens::Error::<T>::NotCrossChainTransferableCurrency
+				);
+			});
+		}
```

Comment:
couldn't those be made used for all runtimes inestead of inside a `mod kusama`?

Reply:
Yes, I think that applies to a lot of these tests tbh. For this one, I wanted to have a 1 to 1 transition more or less.

Reply:
OK, then I'll create a separate module here called `all` and move these over + use all runtimes.

Reply:
Done.